### PR TITLE
Add intermediate count column to show trial progress

### PR DESCRIPTION
**Pull request by lvybriage**
related changes:

- image

- image

- image

- image

### CODE REVIEW

File: src/webui/src/components/Modal/Compare.tsx [hunk 1]
```diff
@@ -83,13 +83,13 @@ class Compare extends React.Component<CompareProps, {}> {
             },
             xAxis: {
                 type: 'category',
-                name: 'Step',
+                // name: '# Intermeidate',
```
Comment (QuanluZhang) [inline src/webui/src/components/Modal/Compare.tsx]: why not just remove this line

File: src/webui/src/components/trial-detail/Intermeidate.tsx [hunk 2]
```diff
@@ -108,7 +108,7 @@ class Intermediate extends React.Component<IntermediateProps, IntermediateState>
                 },
                 xAxis: {
                     type: 'category',
-                    name: 'Step',
+                    // name: '# Intermediate',
```
Comment (QuanluZhang) [inline src/webui/src/components/trial-detail/Intermeidate.tsx]: remove

File: src/webui/src/components/trial-detail/TableList.tsx [hunk 3]
```diff
@@ -415,37 +430,37 @@ class TableList extends React.Component<TableListProps, TableListState> {
...
+                        width: 120,
                         render: (text: string, record: TableObj) => {
                             let trialStatus = record.status;
-                            let flagKill = false;
-                            if (trialStatus === 'RUNNING') {
-                                flagKill = true;
-                            } else {
-                                flagKill = false;
-                            }
+                            const flag: boolean = trialStatus === 'RUNNING' ? false : true;
```
Comment (liuzhe-lz) [inline src/webui/src/components/trial-detail/TableList.tsx]: `const flag: boolean = (trialStatus !== 'RUNNING')`

File: src/webui/src/components/trial-detail/Intermeidate.tsx [hunk 4]
```diff
@@ -282,58 +282,52 @@ class Intermediate extends React.Component<IntermediateProps, IntermediateState>
...
+                                    // placeholder="point"
+                                    ref={input => this.pointInput = input}
+                                    className="strange"
+                                />
+                                <span>Metric range</span>
+                                <input
+                                    // placeholder="range"
+                                    ref={input => this.minValInput = input}
+                                />
+                                <span className="heng">-</span>
```
Comment (liuzhe-lz) [inline src/webui/src/components/trial-detail/Intermeidate.tsx]: hyphen

File: src/webui/src/components/Modal/Compare.tsx (hunk 1 above)
Comment (liuzhe-lz) [reply src/webui/src/components/Modal/Compare.tsx]: This indicates that the xAxis name is '# Intermediate' (logically).
We hide it here to fool the library.

File: src/webui/src/components/trial-detail/TableList.tsx [hunk 5]
```diff
@@ -295,13 +295,15 @@ class TableList extends React.Component<TableListProps, TableListState> {
...
                     // concat default column and finalkeys
                     const item = Object.keys(temp);
-                    const want: Array<string> = [];
-                    Object.keys(item).map(key => {
-                        if (item[key] !== 'default') {
-                            want.push(item[key]);
-                        }
-                    });
-                    showTitle = COLUMN.concat(want);
+                    if (item.length <= 1) {
```
Comment (QuanluZhang) [inline src/webui/src/components/trial-detail/TableList.tsx]: could add comments here to explain why check whether item.length is smaller than 1

File: src/webui/src/components/trial-detail/TableList.tsx [hunk 6]
```diff
@@ -295,13 +295,15 @@ class TableList extends React.Component<TableListProps, TableListState> {
...
+                        const want: Array<string> = [];
+                        Object.keys(item).map(key => {
```
Comment (liuzhe-lz) [inline src/webui/src/components/trial-detail/TableList.tsx]: item.forEach(value => { if (value !== ...

File: src/webui/src/components/trial-detail/TableList.tsx (hunk 3 above)
Comment (QuanluZhang) [inline src/webui/src/components/trial-detail/TableList.tsx]: code refactor, or logic change?

File: src/webui/src/components/trial-detail/TableList.tsx [hunk 7]
```diff
@@ -295,13 +295,15 @@ class TableList extends React.Component<TableListProps, TableListState> {
...
+                            if (item[key] !== 'default') {
+                                want.push(item[key]);
+                            }
+                        });
```
Comment (liuzhe-lz) [inline src/webui/src/components/trial-detail/TableList.tsx]:
```
item.forEach(value => {
    if (value !== 'default') {
        want.push(value);
    }
});
```

File: src/webui/src/components/trial-detail/TableList.tsx (hunk 3 above)
Comment (lvybriage) [reply src/webui/src/components/trial-detail/TableList.tsx]: Ok.

File: src/webui/src/components/trial-detail/Intermeidate.tsx (hunk 4 above)
Comment (lvybriage) [reply src/webui/src/components/trial-detail/Intermeidate.tsx]: Ok.

File: src/webui/src/components/trial-detail/TableList.tsx (hunk 3 above)
Comment (lvybriage) [reply src/webui/src/components/trial-detail/TableList.tsx]: It's code refactor.

File: src/webui/src/components/trial-detail/TableList.tsx (hunk 5 above)
Comment (lvybriage) [reply src/webui/src/components/trial-detail/TableList.tsx]: maybe have some questions... I'll test an experiment with dic final keys

**Pull request merged**